multiview/embed/mvmds.py

```diff
@@ -0,0 +1,239 @@
+
+                    else:
+                        # Refers to d value from previous iteration
+                        s2 = s2 + (d2 * Xs[yy] / d[:, yy])
+
+                # eigenvectors dotted with S matrix and pi
+                w = np.dot(s2, q)
+
+                w = np.dot(pi, w)
+
+                q = w / np.sqrt(np.dot(w.T, w))
+
+                for yy in np.arange(views):
+
+                    d[:, yy] = np.dot(np.dot(q.T, Xs[yy]), q)
+
+            # creates next component
+            components[:, i] = q[:, 0]
+            # initializes pi for next iteration
+            pi = pi - np.dot(q, q.T)
+
+        return(components)
+
+    def fit(self, Xs):
+
+        """
+        Calculates dimensionally reduced components by inputting the Euclidean
+        distances of each view, double centering them, and using the _commonpcs
+        function to find common components between views. Works similarly to
+        traditional, single-view Multidimensional Scaling.
+
+        Parameters
+        ----------
+
+        Xs: list of array-likes or numpy.ndarray
+                - Xs length: n_views
+                - Xs[i] shape: (n_samples, n_features_i)
+
+        Attributes
+        ----------
+
+        components: A k-dimensional projection of shape [n,k]
+
+        """
+
+        if (self.n_components) > len(Xs[0]):
+            self.n_components = len(Xs[0])
+            warnings.warn('The number of components you have requested is '
+                          + 'greater than the number of samples in the '
+                          + 'dataset. ' + str(self.n_components)
+                          + ' components were computed instead.')
+
+        if (self.num_iter) <= 0:
+            raise ValueError('The number of iterations must be greater than 0')
+
+        if (self.n_components) <= 0:
+            raise ValueError('The number of components must be greater than 0 '
+                             + 'and less than the number of features')
+
+        Xs = check_Xs(Xs, multiview=True)
+
+        mat = np.ones(shape=(len(Xs), len(Xs[0]), len(Xs[0])))
+
+        # Double centering each view as in single-view MDS
+        for i in np.arange(len(Xs)):
+            view = euclidean_distances(Xs[i])
+            view_squared = np.power(np.array(view), 2)
+
+            J = np.eye(len(view)) - (1/len(view))*np.ones(view.shape)
+            B = -(1/2) * np.matmul(np.matmul(J, view_squared), J)
+            mat[i] = B
+
+        self.components = self._commonpcs(mat)
+
+    def transform(self, Xs):
+
+        """
+        Returns back the original Xs. Not needed for implementation
+
+        Parameters
+        ----------
+
+        Xs: list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            The data to embed based on the prior fit function.
+
+        Returns
+        -------
+
+        Xs = Same Xs as inputted
```

Comment:
You can remove the transform() function. There is no real reason for keeping this code here so far as i can tell.